test/db.test.js

```diff
@@ -490,6 +490,27 @@ describe('Database', function () {
       })
     })
 
+    it('Can use a compound index to get docs with a basic match', function (done) {
```

Comment:
You are going to hate me, really, but we implement tests both in callback and async versions..

Reply:
@tex0l this is done